private/PROPOSAL/Sector.cxx

```diff
@@ -45,7 +45,7 @@ Sector::Definition::Definition()
     , location(Sector::ParticleLocation::InsideDetector)
     , utility_def()
     , cut_settings()
-    , medium_(new Ice())
+    , medium_(std::make_shared<const Medium>(Ice()))
```

Comment:
This is stack allocating a Ice and passing this to the copy constructor of `Medium`.

This should be `std::make_shared<const Ice>()`

Reply:
Thank you, I was not aware of the implicit conversion in smart pointers.